exam/exam.py

```diff
@@ -225,7 +227,32 @@ def on_change_student(self,student, exam_id, standard_id):
         val.update({'standard_id' : student_data.standard_id.id,
                     'roll_no_id' : student_data.roll_no})
         return {'value': val}
-
+    
+    @api.onchange('third_party')
+    def check_category_for_third_party(self):
+        domain = {'domain':{'leads_category':[('third_party','=',False)]}}
+        if self.third_party:
+            domain = {'domain':{'leads_category':[('third_party','=',True)]}}
+        return domain
+    
```

Comment:
Rewrite in one line like:
def check...(self)
    return {'domain':{'leads_category':[('third_party','=',self.third_party or False)]}}
Make things simple, don't complicate.